### PR TITLE
Pass FEATURE_GATES flag to e2e tests

### PR DESCRIPTION
**Pull request by akremsa**
This PR is a part of #1411 and allows to turn on\off feature gates behaviour in e2e tests. 

Also updated some tests introduced in #1396

### CODE REVIEW

File: test/e2e/main_test.go [hunk 1]
```diff
@@ -28,6 +29,8 @@ const defaultNs = "default"
 var framework *e2eframework.Framework
 
 func TestMain(m *testing.M) {
+	runtime.FeatureTestMutex.Lock()
```
Comment (aLekSer) [inline test/e2e/main_test.go:32]: Overall looks great. Can you please move this closer to where it actually used

**Comment by aLekSer** [comment]
I think we need to update `build/e2e-image/e2e.sh`
Or add default parameter :
```
test-e2e: FEATURE_GATES ?= "PlayerCount=true&ContainerPort=true"
test-e2e:
	$(MAKE) test-e2e-integration
	$(MAKE) test-e2e-failure
```

**Comment by akremsa** [comment]
> FEATURE_GATES

Moved FEATURE_GATES the the top of Makefile

File: test/e2e/main_test.go [hunk 2]
```diff
@@ -50,6 +51,13 @@ func TestMain(m *testing.M) {
 		log.Printf("failed to cleanup resources: %v\n", err)
 	}
 
+	runtime.FeatureTestMutex.Lock()
```
[the remark this answers is not in the document]
Comment (akremsa) [reply test/e2e/main_test.go:54]: Ok, I see

[the remark this answers is not in the document]
Comment (akremsa) [reply test/e2e/main_test.go]: Fixed

File: test/e2e/framework/framework.go [hunk 3]
```diff
@@ -117,6 +118,7 @@ func NewFromFlags() (*Framework, error) {
 	stressTestLevel := flag.Int("stress", 0, "enable stress test at given level 0-100")
 	perfOutputDir := flag.String("perf-output", "", "write performance statistics to the specified directory")
 	version := flag.String("version", "", "agones controller version to be tested, consists of release version plus a short hash of the latest commit")
+	featureGates := flag.String("featuregates", "", "list of feature gates to be tested in the test run")
```
Comment (markmandel) [inline test/e2e/framework/framework.go:121]: Should this be the same as https://github.com/googleforgames/agones/blob/master/pkg/util/runtime/features.go#L66 ?

We have `featureGateFlag` variable set in there - would make sense to keep it all the same. Then we can reuse [ParseFeaturesFromEnv](https://github.com/googleforgames/agones/blob/master/pkg/util/runtime/features.go#L79) as well.

(Long term, should we move this to pflag?)

**Comment by akremsa** [comment]
@markmandel  I've added pflag and viper to e2e framework package but there is an issue:
m.Run() in TestMain uses flag.Parse() from the standard go package and it raises an error _"flag provided but not defined"_ because standard flag package doesn't consider a flag to be declared as a variable if it is registered using pflag. 
I've used a workaround which is discussed here:
https://github.com/spf13/pflag/issues/63

New `func SplitFlags()` should be used every time we want to parse flags before running tests.

In my opinion it brings redundant complexity to the code, but if you think that adding viper to tests worth that - we can keep it.
What do you think?

**Comment by aLekSer** [comment]
Seems to be a new one:
```
{"message":"agones.dev/agones/pkg/client/informers/externalversions/factory.go:119: Failed to watch *v1.GameServer: unhandled watch: testing.WatchActionImpl{ActionImpl:testing.ActionImpl{Namespace:\"default\", Verb:\"watch\", Resource:schema.GroupVersionResource{Group:\"agones.dev\", Version:\"v1\", Resource:\"gameservers\"}, Subresource:\"\"}, WatchRestrictions:testing.WatchRestrictions{Labels:labels.internalSelector(nil), Fields:(*fields.hasTerm)(0xc00098cc60), ResourceVersion:\"\"}}","severity":"error","time":"2020-04-04T13:21:34.586270946Z"}
{"gsKey":"default/test","message":"Created GameServer sidecar","severity":"info","source":"*sdkserver.SDKServer","time":"2020-04-04T13:21:34.577604883Z"}
        	Test:       	TestLocalSDKServerPlayerCapacity
        	Error:      	timeout getting watch
        	Error Trace:	localsdk_test.go:312
    localsdk_test.go:312: 
--- FAIL: TestLocalSDKServerPlayerCapacity (10.07s)
```

File: test/e2e/framework/framework.go [hunk 4]
```diff
@@ -105,26 +110,76 @@ func newFramework(kubeconfig string, qps float32, burst int) (*Framework, error)
...
 }
 
+const (
+	kubeconfigFlag      = "kubeconfig"
+	gsimageFlag         = "gameserver-image"
+	pullSecretFlag      = "pullsecret"
+	stressTestLevelFlag = "stress"
+	perfOutputDirFlag   = "perf-output"
+	versionFlag         = "version"
+	featureGatesFlag    = "feature-gates"
```
Comment (markmandel) [inline test/e2e/framework/framework.go:120]: Curious - why are we re-declaring `featureGatesFlag`? We could use the const value from utils/runtime no?

File: test/e2e/framework/framework.go [hunk 5]
```diff
@@ -105,26 +110,76 @@ func newFramework(kubeconfig string, qps float32, burst int) (*Framework, error)
...
+	kubeconfig := pflag.String(kubeconfigFlag, viper.GetString(kubeconfigFlag), "kube config path, e.g. $HOME/.kube/config")
+	gsimage := pflag.String(gsimageFlag, viper.GetString(gsimageFlag), "gameserver image to use for those tests, gcr.io/agones-images/udp-server:0.19")
+	pullSecret := pflag.String(pullSecretFlag, viper.GetString(pullSecretFlag), "optional secret to be used for pulling the gameserver and/or Agones SDK sidecar images")
+	stressTestLevel := pflag.Int(stressTestLevelFlag, viper.GetInt(stressTestLevelFlag), "enable stress test at given level 0-100")
+	perfOutputDir := pflag.String(perfOutputDirFlag, viper.GetString(perfOutputDirFlag), "write performance statistics to the specified directory")
+	version := pflag.String(versionFlag, viper.GetString(versionFlag), "agones controller version to be tested, consists of release version plus a short hash of the latest commit")
+	runtime.FeaturesBindFlags()
+	pflag.Parse()
+
+	viper.SetEnvKeyReplacer(strings.NewReplacer("-", "_"))
```
Comment (markmandel) [inline test/e2e/framework/framework.go:168]: Having env binding is nice, but do we need it for e2e tests?

File: test/e2e/controller/main_test.go [hunk 6]
```diff
@@ -15,12 +15,12 @@
 package controller
 
 import (
-	"log"
 	"os"
 	"testing"
 
 	e2eframework "agones.dev/agones/test/e2e/framework"
 	"github.com/sirupsen/logrus"
+	log "github.com/sirupsen/logrus"
```
Comment (markmandel) [inline test/e2e/controller/main_test.go:23]: So this is a several cross cutting pieces in one PR.
- Change logging to to Logrus
- Implement pflag over flag (I think I mentioned we should tackle this at a later date?)
- Pass Feature_Gates

For future references, these should be separate PRs, having this all together makes things harder to review and track the changes.

File: test/e2e/framework/framework.go [hunk 7]
```diff
@@ -105,26 +110,76 @@ func newFramework(kubeconfig string, qps float32, burst int) (*Framework, error)
...
+	pullSecretFlag      = "pullsecret"
+	stressTestLevelFlag = "stress"
+	perfOutputDirFlag   = "perf-output"
+	versionFlag         = "version"
+	featureGatesFlag    = "feature-gates"
+)
+
+// SplitFlags This workaround allows to use flag.Parse() in m.Run and pflag.Parse() together
+// https://github.com/spf13/pflag/issues/63
+func SplitFlags() error {
```
Comment (markmandel) [inline test/e2e/framework/framework.go:125]: I'm not sure I understand why this is here, I couldn't find anywhere else where we use the `flag` library other than in this file? And you've now removed all of that, so do we still need this?

File: build/Makefile [hunk 8]
```diff
@@ -246,7 +248,8 @@ test-e2e-integration: $(ensure-build-image)
 	echo "Starting e2e integration test!"
 	$(GO_TEST) $(ARGS) $(agones_package)/test/e2e $(GO_E2E_TEST_ARGS) \
 		--gameserver-image=$(GS_TEST_IMAGE) \
-		--pullsecret=$(IMAGE_PULL_SECRET)
+		--pullsecret=$(IMAGE_PULL_SECRET) \
+		--feature-gates=$(FEATURE_GATES)
```
Comment (markmandel) [inline build/Makefile:252]: Happy to see this match the flags we use elsewhere :+1:

File: test/e2e/framework/framework.go (hunk 4 above)
Comment (akremsa) [reply test/e2e/framework/framework.go:120]: Yes, we can use that variable. Made it exportable.

File: test/e2e/framework/framework.go (hunk 5 above)
Comment (akremsa) [reply test/e2e/framework/framework.go:168]: At this moment - no. I will remove it then.

File: test/e2e/framework/framework.go (hunk 7 above)
Comment (akremsa) [reply test/e2e/framework/framework.go:125]: Yes, we do need this because of:
image

Otherwise, I get an error like _"flag provided but not defined"_.
We do not use viper & pflag in tests that is why there is no such an issue in other places. I didn't find another solution except this.

File: test/e2e/controller/main_test.go (hunk 6 above)
Comment (akremsa) [reply test/e2e/controller/main_test.go:23]: I agree. That was a misunderstanding from my side, I considered your note as "nice to add" in a current scope of changes.

File: test/e2e/framework/framework.go (hunk 7 above)
Comment (aLekSer) [reply test/e2e/framework/framework.go:124]: All kind of `go test -v -run TestOne` should be passed to m.Run() unchanged. Them are prefixed with `test.` and could be ignored by `pflag` but needed for test tool. Without this or similar workaround we could not use pflag with testing packages.

File: test/e2e/main_test.go [hunk 9]
```diff
@@ -39,24 +39,29 @@ func TestMain(m *testing.M) {
 		exitCode int
 	)
 
+	if err = e2eframework.SplitFlags(); err != nil {
+		log.Error("Error parsing regular test flags:", err)
```
Comment (aLekSer) [inline test/e2e/main_test.go:43]: You miss blank space here:
```suggestion
		log.Errorf("Error parsing regular test flags: %v", err)
```

File: test/e2e/main_test.go [hunk 10]
```diff
@@ -39,24 +39,29 @@ func TestMain(m *testing.M) {
...
+		os.Exit(1)
+	}
+
 	if framework, err = e2eframework.NewFromFlags(); err != nil {
-		log.Printf("failed to setup framework: %v\n", err)
+		log.Errorf("failed to setup framework: %v\n", err)
```
Comment (aLekSer) [inline test/e2e/main_test.go:48]: new line is not needed here and below
```suggestion
		log.Errorf("failed to setup framework: %v", err)
```

File: test/e2e/controller/main_test.go [hunk 11]
```diff
@@ -39,21 +39,26 @@ func TestMain(m *testing.M) {
 		exitCode int
 	)
 
+	if err = e2eframework.SplitFlags(); err != nil {
+		log.Error("Error parsing regular test flags:", err)
```
Comment (aLekSer) [inline test/e2e/controller/main_test.go:43]:
```suggestion
		log.Errorf("Error parsing regular test flags: %v", err)
```

**Comment by aLekSer** [comment]
Error in SDK conformance test for C#:
```
make[1]: Entering directory '/workspace/build'
includes/sdk.mk:85: recipe for target 'run-sdk-command-csharp' failed
make: *** Waiting for unfinished jobs....
make: *** [run-sdk-command-csharp] Error 2
make[1]: *** [run-sdk-command] Error 1
includes/sdk.mk:89: recipe for target 'run-sdk-command' failed
/usr/share/dotnet/sdk/2.2.402/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(129,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [/go/src/agones.dev/agones/sdks/csharp/sdk/csharp-sdk.csproj]
/usr/share/dotnet/sdk/2.2.402/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.targets(129,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [/go/src/agones.dev/agones/sdks/csharp/sdk/csharp-sdk.csproj]
```

File: test/e2e/framework/framework.go (hunk 7 above)
Comment (markmandel) [reply test/e2e/framework/framework.go:124]: oooh! It gets called in the Go Test code? I would never have thought of that.

This issue was from 2016, I take there's been no fix since then?

Comment (markmandel) [reply test/e2e/framework/framework.go:124]: https://github.com/spf13/pflag#supporting-go-flags-when-using-pflag seems relevent?

Comment (akremsa) [reply test/e2e/framework/framework.go]: Just tried it - works perfectly.

Comment (aLekSer) [reply test/e2e/framework/framework.go]: Also seen this function before, but this readme section better explains how to use it.

Comment (aLekSer) [reply test/e2e/framework/framework.go]: Well it seems that KeepGoFlags feature are not yet in place, so this workaround is needed. 
https://github.com/spf13/pflag/issues/63#issuecomment-215143145
And all `go test` flags are ignored by `pflag` lib. @akremsa  has just tested that `-run`, `-v` are skipped.

Comment (aLekSer) [reply test/e2e/framework/framework.go]: There is a more recent bug with `-test.v` flag
https://github.com/spf13/pflag/issues/238

Comment (akremsa) [reply test/e2e/framework/framework.go]: Yes, a bit tricky issue. We fixed passing custom flags by adding `pflag.CommandLine.AddGoFlagSet(flag.CommandLine)`, but broke passing golang test flags.
https://github.com/spf13/pflag/issues/238

I suggest to revert changes back and leave SplitFlags() function.

File: test/e2e/controller/main_test.go [hunk 12]
```diff
@@ -40,20 +40,20 @@ func TestMain(m *testing.M) {
 	)
 
 	if framework, err = e2eframework.NewFromFlags(); err != nil {
-		log.Printf("failed to setup framework: %v\n", err)
+		log.Errorf("failed to setup framework: %v", err)
```
Comment (markmandel) [inline test/e2e/controller/main_test.go:43]: if we are using logrus, then we should do:

```suggestion
		log.WithError(err).Error("failed to setup framework")
```

No reason to use Errorf like before. Let's go all in. Same with below.

File: test/e2e/framework/framework.go [hunk 13]
```diff
@@ -105,26 +108,53 @@ func newFramework(kubeconfig string, qps float32, burst int) (*Framework, error)
...
+	pflag.Parse()
+
+	runtime.Must(viper.BindEnv(kubeconfigFlag))
+	runtime.Must(viper.BindEnv(gsimageFlag))
+	runtime.Must(viper.BindEnv(pullSecretFlag))
+	runtime.Must(viper.BindEnv(stressTestLevelFlag))
+	runtime.Must(viper.BindEnv(perfOutputDirFlag))
+	runtime.Must(viper.BindEnv(versionFlag))
+	runtime.Must(viper.BindPFlags(pflag.CommandLine))
+	runtime.Must(runtime.FeaturesBindEnv())
```
Comment (markmandel) [inline test/e2e/framework/framework.go:151]: :raised_hands:

File: test/e2e/main_test.go [hunk 14]
```diff
@@ -40,23 +40,23 @@ func TestMain(m *testing.M) {
 	)
 
 	if framework, err = e2eframework.NewFromFlags(); err != nil {
-		log.Printf("failed to setup framework: %v\n", err)
+		log.Errorf("failed to setup framework: %v", err)
```
Comment (markmandel) [inline test/e2e/main_test.go:43]: Nit: Same as above, let's with `WithError`

File: pkg/util/runtime/features.go [hunk 15]
```diff
@@ -25,7 +25,8 @@ import (
 )
 
 const (
-	featureGateFlag = "feature-gates"
+	// FeatureGateFlag is a name of a command line flag
```
Comment (aLekSer) [inline pkg/util/runtime/features.go:28]:
```suggestion
	// FeatureGateFlag is a name of a command line flag, which turns on specific tests for FeatureGates
```

File: test/e2e/framework/framework.go [hunk 16]
```diff
@@ -105,26 +110,67 @@ func newFramework(kubeconfig string, qps float32, burst int) (*Framework, error)
...
+	stressTestLevelFlag = "stress"
+	perfOutputDirFlag   = "perf-output"
+	versionFlag         = "version"
+)
+
+// ParseTestFlags Parses go test flags separately because pflag package ignores flags with '-test.' prefix
+// Related issues:
+// https://github.com/spf13/pflag/issues/63
+// https://github.com/spf13/pflag/issues/238
+func ParseTestFlags() error {
```
Comment (aLekSer) [inline test/e2e/framework/framework.go:126]: It is a bit hacky way, but since this is for test and we have not send a PR to spf13 repo this is an acceptable and small workaround.

**Comment by aLekSer** [comment]
A new error in SDK conformance `/bin/sh: 2: -e: not found`:
```
/usr/bin/make ensure-image IMAGE_TAG=gcr.io/agones-images/agones-sdk:1.6.0-1047ab7 BUILD_TARGET=build-agones-sdk-server-image
Step #22 - "sdk-conformance": DOCKER_RUN_ARGS="--net host -e AGONES_SDK_GRPC_PORT=9357 -e AGONES_SDK_HTTP_PORT=9358 -e FEATURE_GATES="PlayerTracking=true&ContainerPortAllocation=true" " COMMAND=sdktest /usr/bin/make run-sdk-command & \
Step #22 - "sdk-conformance": docker run -p 9357:9357 -p 9358:9358 -e "FEATURE_GATES="PlayerTracking=true&ContainerPortAllocation=true"" -e "ADDRESS=" -e "TEST=ready,allocate,setlabel,setannotation,gameserver,health,shutdown,watch,reserve" -e "TIMEOUT=30" -e "DELAY=3" \
Step #22 - "sdk-conformance": --net=host gcr.io/agones-images/agones-sdk:1.6.0-1047ab7 --grpc-port 9357 --http-port 9358
Step #22 - "sdk-conformance": /bin/sh: 2: -e: not found
Step #22 - "sdk-conformance": includes/sdk.mk:145: recipe for target 'run-sdk-conformance-no-build' failed
Step #22 - "sdk-conformance": make[2]: *** [run-sdk-conformance-no-build] Error 127
Step #22 - "sdk-conformance": make[1]: *** [run-sdk-conformance-test] Error 2
Step #22 - "sdk-conformance": includes/sdk.mk:152: recipe for target 'run-sdk-conformance-test' failed
Step #22 - "sdk-conformance": make: *** [run-sdk-conformance-test-rust] Error 2
Step #22 - "sdk-conformance": includes/sdk.mk:168: recipe for target 'run-sdk-conformance-test-rust' failed
```

**Comment by aLekSer** [comment]
@markmandel as soon as there is nothing else to change, we could merge this PR?

**Comment by aLekSer** [comment]
@akremsa Well, it seems that moving `FEATURE_GATES` up to a global level results in failing SDK conformance tests. What I would recommend is to change `sdk.mk` and set FEATURE_GATES there also,for some test it should be "". Or we should specify this variable for E2E tests and install separately.
```
Step #22 - "sdk-conformance": DOCKER_RUN_ARGS="--net host -e AGONES_SDK_GRPC_PORT=9002 -e AGONES_SDK_HTTP_PORT=9102 -e FEATURE_GATES="PlayerTracking=true&ContainerPortAllocation=true" " COMMAND=sdktest /usr/bin/make run-sdk-command & \
Step #22 - "sdk-conformance": docker run -p 9002:9002 -p 9102:9102 -e "FEATURE_GATES="PlayerTracking=true&ContainerPortAllocation=true"" -e "ADDRESS=" -e "TEST=ready,allocate,setlabel,setannotation,gameserver,health,shutdown,watch,reserve" -e "SDK_NAME=node" -e "TIMEOUT=30" -e "DELAY=3" \
```

**Comment by akremsa** [comment]
> @akremsa Well, it seems that moving `FEATURE_GATES` up to a global level results in failing SDK conformance tests. What I would recommend is to change `sdk.mk` and set FEATURE_GATES there also,for some test it should be "". Or we should specify this variable for E2E tests and install separately.
> 
> ```
> Step #22 - "sdk-conformance": DOCKER_RUN_ARGS="--net host -e AGONES_SDK_GRPC_PORT=9002 -e AGONES_SDK_HTTP_PORT=9102 -e FEATURE_GATES="PlayerTracking=true&ContainerPortAllocation=true" " COMMAND=sdktest /usr/bin/make run-sdk-command & \
> Step #22 - "sdk-conformance": docker run -p 9002:9002 -p 9102:9102 -e "FEATURE_GATES="PlayerTracking=true&ContainerPortAllocation=true"" -e "ADDRESS=" -e "TEST=ready,allocate,setlabel,setannotation,gameserver,health,shutdown,watch,reserve" -e "SDK_NAME=node" -e "TIMEOUT=30" -e "DELAY=3" \
> ```
> 
> Previously with Successful SDK test runs this parameter was an empty string:
> 
> ```
> DOCKER_RUN_ARGS="--net host -e AGONES_SDK_GRPC_PORT=9002 -e AGONES_SDK_HTTP_PORT=9102 -e FEATURE_GATES= " COMMAND=sdktest /usr/bin/make run-sdk-command & \
> Step #22 - "sdk-conformance": docker run -p 9002:9002 -p 9102:9102 -e "FEATURE_GATES=" -e "ADDRESS=" -e "TEST=ready,allocate,setlabel,setannotation,gameserver,health,shutdown,watch,reserve" -e "SDK_NAME=node" -e "TIMEOUT=30" -e "DELAY=4" \
> Step #22 - "sdk-conformance": --net=host gcr.io/agones-images/agones-sdk:1.6.0-75ba12c --grpc-port 9002 --http-port 9102
> Step #22 - "sdk-conformance": DOCKER_RUN_ARGS="--net host -e AGONES_SDK_GRPC_PORT=9003 -e AGONES_SDK_HTTP_PORT=9103 -e FEATURE_GATES= " COMMAND=sdktest /usr/bin/make run-sdk-command 
> ```

Fixed

File: build/includes/sdk.mk [hunk 17]
```diff
@@ -139,7 +139,7 @@ run-sdk-conformance-no-build: DELAY ?= $(shell bash -c "echo $$[ ($(RANDOM) % 5
 run-sdk-conformance-no-build: TESTS ?= $(DEFAULT_CONFORMANCE_TESTS)
 run-sdk-conformance-no-build: GRPC_PORT ?= 9357
 run-sdk-conformance-no-build: HTTP_PORT ?= 9358
-run-sdk-conformance-no-build: FEATURE_GATES ?=
+run-sdk-conformance-no-build: FEATURE_GATES := ""
```
Comment (markmandel) [inline build/includes/sdk.mk:142]: Curious - why this change? Don't we want it to be overwritable?

File: pkg/util/runtime/features.go [hunk 18]
```diff
@@ -25,7 +25,8 @@ import (
 )
 
 const (
-	featureGateFlag = "feature-gates"
+	// FeatureGateFlag is a name of a command line flag, which turns on specific tests for FeatureGates
+	FeatureGateFlag = "feature-gates"
```
Comment (markmandel) [inline pkg/util/runtime/features.go:29]: Nice! :+1:

File: build/includes/sdk.mk (hunk 17 above)
Comment (akremsa) [reply build/includes/sdk.mk:142]: I consider it as the easiest solution at the moment. Probably it is worth adding a comment as we do in _terraform.mk_ - "All Feature gates are disabled". What do you think?

Comment (markmandel) [reply build/includes/sdk.mk:142]: But we need to be able to pass feature flags to conformance tests. We can't remove that ability. In fact, this will turn off tests for the Go conformance tests, so we can't do this.

We have to use `?=`, so that it can be overwritten.

File: build/Makefile [hunk 19]
```diff
@@ -59,6 +59,8 @@ KIND_CONTAINER_NAME=$(KIND_PROFILE)-control-plane
 # Game Server image to use while doing end-to-end tests
 GS_TEST_IMAGE ?= gcr.io/agones-images/udp-server:0.19
 
+FEATURE_GATES ?= "PlayerTracking=true&ContainerPortAllocation=true"
```
Comment (markmandel) [inline build/Makefile:62]: Maybe this should be `ALL_FEATURE_GATES` so that then for targets that want to default to full feature games one could do:

`target: FEATURE_GATE ?= $(ALL_FEATURE_GATES)`

And it won't conflict with anything it shouldn't.

File: build/includes/sdk.mk (hunk 17 above)
Comment (markmandel) [reply build/includes/sdk.mk:142]: https://www.gnu.org/software/make/manual/html_node/Setting.html 

If you want to understand the different assignments in Make.

File: build/Makefile (hunk 19 above)
Comment (akremsa) [reply build/Makefile:62]: Ok, I will try this solution.

Comment (akremsa) [reply build/Makefile]: @markmandel done. Please check.

File: build/Makefile [hunk 20]
```diff
@@ -288,7 +291,6 @@ install: PING_SERVICE_TYPE := "LoadBalancer"
 install: ALLOCATOR_SERVICE_TYPE := "LoadBalancer"
 install: CRD_CLEANUP := true
 install: LOG_LEVEL := "debug"
-install: FEATURE_GATES := "PlayerTracking=true&ContainerPortAllocation=true"
```
Comment (aLekSer) [inline build/Makefile:291]: This line should be left as it was, but with FEATURE_GATES ?= ALL_FEATURE_GATES

File: build/Makefile [hunk 21]
```diff
@@ -246,7 +248,8 @@ test-e2e-integration: $(ensure-build-image)
 	echo "Starting e2e integration test!"
```
Comment (aLekSer) [inline build/Makefile:248]: Add `FEATURE_GATES ?= ALL_FEATURE_GATES` here or above. So we can change them for specific

File: build/Makefile (hunk 20 above)
Comment (aLekSer) [reply build/Makefile:291]: Now it is better)
